cpp/src/db/db_object.cu

```diff
@@ -120,84 +120,62 @@ bool db_pattern<idx_t>::isAllConstants() {
 template class db_pattern<int32_t>;
 template class db_pattern<int64_t>;
 
-template<typename idx_t>
-void db_column_index<idx_t>::deleteData() {
-  if (offsets != nullptr) {
-    ALLOC_FREE_TRY(offsets, nullptr);
-    offsets = nullptr;
-    offsets_size = 0;
-  }
-  if (indirection != nullptr) {
-    ALLOC_FREE_TRY(indirection, nullptr);
-    indirection = nullptr;
-    indirection_size = 0;
-  }
-}
-
 template<typename idx_t>
 db_column_index<idx_t>::db_column_index() {
-  offsets = nullptr;
   offsets_size = 0;
-  indirection = nullptr;
   indirection_size = 0;
 }
 
 template<typename idx_t>
-db_column_index<idx_t>::db_column_index(idx_t* _offsets,
+db_column_index<idx_t>::db_column_index(rmm::device_buffer&& _offsets,
                                         idx_t _offsets_size,
-                                        idx_t* _indirection,
+                                        rmm::device_buffer&& _indirection,
                                         idx_t _indirection_size) {
-  offsets = _offsets;
+  offsets = std::move(_offsets);
   offsets_size = _offsets_size;
-  indirection = _indirection;
+  indirection = std::move(_indirection);
   indirection_size = _indirection_size;
 }
 
 template<typename idx_t>
 db_column_index<idx_t>::db_column_index(db_column_index<idx_t>&& other) {
-  offsets = other.offsets;
+  offsets = std::move(other.offsets);
   offsets_size = other.offsets_size;
-  indirection = other.indirection;
+  indirection = std::move(other.indirection);
   indirection_size = other.indirection_size;
-  other.offsets = nullptr;
   other.offsets_size = 0;
-  other.indirection = nullptr;
   other.indirection_size = 0;
 }
```

Comment:
db_column_index(db_column_index<idx_t>&& other) = default; should be sufficient.